### PR TITLE
Return region filter information from R API

### PR DESCRIPTION
**Pull request by r-ash**
This PR will

* Return region filter hierarchy from geojson
* Also returns some metadata about region level and label which front end isn't currently using but will need at some point

Looks like the INLA build is working on this branch so not sure what happened on master

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -44,3 +44,9 @@ collapse <- function(vector, collapse = ", ", limit = 150, end = "...") {
   out <- paste(vector, collapse = collapse)
   out
 }
+
+`%||%` <- function(a, b) if (is.null(a)) b else a
```
Comment (richfitz) [inline R/utils.R]: could you reformat this (this is my fault)

File: tests/testthat/test-filters.R [hunk 2]
```diff
@@ -54,3 +54,100 @@ test_that("get_survey_filters gets available filter options and sorts them", {
   expect_equal(get_survey_filters(NULL), list())
   expect_equal(get_age_filters(data.frame(survey_id = NULL)), list())
 })
+
+test_that("can construct sorted tree from data frame", {
+  data <- data.frame(
+    id = c("MWI", "MWI.1", "MWI.2", "MWI.1.1", "MWI.1.2"),
+    parent_id = c(NA, "MWI", "MWI", "MWI.1", "MWI.1"),
+    sort_order = c(1, 2, 3, 4, 5),
+    stringsAsFactors = FALSE
```
Comment (richfitz) [inline tests/testthat/test-filters.R]: you could use `data_frame` and reduce noise here

File: inst/schema/ValidateInputResponse.schema.json [hunk 3]
```diff
@@ -25,9 +25,13 @@
...
-          "properties" :{
-            "regions": {
+          "properties" : {
+            "level_labels": {
               "type": "array",
+              "items": { "$ref": "LevelLabels.schema.json"}
+            },
+            "regions": {
+              "type": "object",
               "items": { "$ref": "NestedFilterOption.schema.json" }
```
Comment (richfitz) [inline inst/schema/ValidateInputResponse.schema.json]: It looks like this schema does not match the data:

```
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "type": "object",
  "properties": {
    "name": {
      "type": "string"
    },
    "id": {
      "type": "string"
    },
    "options": {
      "type": "array",
      "items": {
        "type": "object"
      }
    }
  },
  "additionalProperties": false,
  "required": [
    "name", "id"
  ]
}
```

it should be recursive, right?

Comment (r-ash) [reply inst/schema/ValidateInputResponse.schema.json]: Fixed that.

File: tests/testthat/test-filters.R (hunk 2 above)
Comment (r-ash) [reply tests/testthat/test-filters.R]: Nice - have fixed that throughout.

File: R/utils.R (hunk 1 above)
Comment (r-ash) [reply R/utils.R]: Yeah sorry, I had said I would do that and then subsequently forgot about it. Tidied up now.

**Pull request merged**